cmd/deploy/main.go

```diff
@@ -217,10 +220,13 @@ func main() {
 							panic(err) // TODO exit in a more clean way (we can't use "os.Exit" because that causes *more* errors 😭)
 						}
 						dryRunOut <- j
+
+						// TODO this isn't exactly *failure* -- maybe we should just add a logText suffix instead?
+						fmt.Fprintln(os.Stderr, failurePrefix+logText)
```

Comment:
A good argument in favor of dropping the `TODO` here and just accepting that this *is* failure is that this case is basically "we do need to push, but were blocked by configuration"

Reply:
I agree that an unpushed image should be considered a failure on `dry-run`. Should this also exit at the end with an error if there were things needing to be pushed when doing `dry-run`?

Reply:
That's a harder question -- I think it's more useful/easier to use if "non-empty stdout" means something needed to be pushed, since the whole point is querying what _did_ need to be pushed, and a non-zero exit code then makes it harder to differentiate breaking errors vs just "things need to be pushed".